__tests__/app.comments.test.js

```diff
@@ -36,31 +43,138 @@ describe("GET", () => {
+  });
+  test(`400: responds with an error message if the article_id is not valid`, () => {
+    return request(app)
+      .get(`/api/articles/three/comments`)
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe(`Invalid input`);
+      });
+  });
+});
+
+describe("POST - /api/articles/:article_id/comments", () => {
+  test(`201: responds with an object {comment: comment}, where the value 'comment' is the posted comment`, () => {
+    const comment = {
+      username: "butter_bridge",
+      body: "Quite a lot of tickets in the project",
+    };
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(comment)
+      .expect(201)
+      .then(({body}) => {
+        expect(body.comment).toEqual(
+          expect.objectContaining({
+            article_id: 4,
+            author: comment.username,
+            body: comment.body,
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+  test("201: inserts a posted comment in the database", () => {
+    const comment = {
+      username: "butter_bridge",
+      body: "Quite a lot of tickets in the project",
+    };
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(comment)
+      .expect(201)
+      .then(({body}) => {
+        expect(body.comment).toEqual(
+          expect.objectContaining({
+            comment_id: 19,
+            votes: 0,
+          })
+        );
+        return db.query(`SELECT * FROM comments;`);
+      })
+      .then(({rowCount}) => {
+        expect(rowCount).toBe(19);
+      });
+  });
+  test(`400: returns an error message if 'comment body' on the request body is missed`, () => {
+    const commentWithoutBody = {
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(commentWithoutBody)
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe("Required fields are missed");
+      });
+  });
+  test(`400: returns an error message if 'comment username' on the request body is missed`, () => {
+    const commentWithoutBody = {
+      body: "Coding is fun",
+    };
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(commentWithoutBody)
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe("Required fields are missed");
+      });
+  });
+  test(`400: returns an error message if the username doesn't exist in the database`, () => {
+    const commentFromNotSighedUpPerson = {
+      username: "not_signed_up",
+      body: "It takes longer to write tests than to write code",
+    };
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(commentFromNotSighedUpPerson)
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe("User with this name does not exist");
+      });
+  });
+  test(`400: returns an error message if the article_id doesn't exist in the database but is valid`, () => {
```

Comment:
THis test would be a 404 not found as the article can't be found

Reply:
Why for username which also doesn't exist in database status 400 is good but here is bad?